src/components/SurveyDetails/index.js

```diff
@@ -56,8 +56,8 @@ class SurveyDetails extends Component {
         this.props.requestAddress(id);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.saving && !nextProps.saving) {
+    componentDidUpdate(prevProps) {
+        if (prevProps.saving && !this.props.saving) {
```

Comment:
hacelo consistente y elegí si primero evaluas `this.props` o `prevProps` en todos los `componentDidUpdate`.